app/src/main/java/org/p2p/wallet/send/model/SendSolanaFee.kt

```diff
@@ -177,9 +190,18 @@ data class SendSolanaFee constructor(
                 appendLine("alternativeFeePayerTokens = ${alternativeFeePayerTokens.map(Token.Active::tokenSymbol)}")
                 appendLine("isValidToSwitchOnSource = $isValidToSwitchOnSource")
                 appendLine("shouldSwitchToSpl = $shouldSwitchToSpl")
+                appendLine("feePayerToken = ${feePayerToken.tokenSymbol}")
+                appendLine("feePayerTokenCanCoverExpenses = $feePayerTokenCanCoverExpenses")
             }
         )
         return when {
+            feePayerTokenCanCoverExpenses -> {
+                if (feePayerToken.isSOL) {
+                    SwitchToSol
+                } else {
+                    SwitchToSpl(feePayerToken)
+                }
+            }
```

Comment:
I don't want to add a new useless state like `NoSwitch`, but it might be a better choise..